core/src/main/java/com/sap/ai/sdk/core/AiCoreService.java

```diff
@@ -161,8 +161,7 @@ public AiCoreDeployment forDeploymentByScenario(@Nonnull final String scenarioId
   @Nonnull
   protected Destination getBaseDestination()
       throws DestinationAccessException, DestinationNotFoundException {
-    Dotenv.configure().ignoreIfMissing().load();
-    val serviceKey = System.getenv("AICORE_SERVICE_KEY");
+    val serviceKey = Dotenv.configure().ignoreIfMissing().load().get("AICORE_SERVICE_KEY");
```

Comment:
(Comment)

Please be aware, `getBaseDestination()` is uncached and will invoke an I/O heavy operation to parse a local text file with every core class interaction by default.